# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/mivisionx/hotfix-opencv4.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- MIVisionX-rocm-6.3.3/utilities/runvx/CMakeLists.txt.vanilla	2025-03-28 19:20:24.952645318 +0100
+++ MIVisionX-rocm-6.3.3/utilities/runvx/CMakeLists.txt	2025-03-28 19:21:24.416072848 +0100
@@ -116,7 +116,7 @@
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DHAVE_OpenSSL=1")
         target_link_libraries(runvx crypto)
     endif()
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-declarations -std=gnu++14")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-declarations -std=gnu++14 -I/usr/include/opencv4")
 endif()
 
 # install MIVisionX executables -- {ROCM_PATH}/bin
--- MIVisionX-rocm-6.3.3/amd_openvx_extensions/amd_opencv/CMakeLists.txt.vanilla	2025-03-28 19:27:26.415939874 +0100
+++ MIVisionX-rocm-6.3.3/amd_openvx_extensions/amd_opencv/CMakeLists.txt	2025-03-28 19:27:53.118678056 +0100
@@ -128,7 +128,7 @@
 else()
 	target_link_libraries(vx_opencv openvx)
 	# -fdata-sections -- Place data items into their own section.
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fdata-sections -std=gnu++14")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fdata-sections -std=gnu++14 -I/usr/include/opencv4")
 endif()
 
 message("-- ${White}vx_opencv -- CMAKE_CXX_FLAGS:${CMAKE_CXX_FLAGS}${ColourReset}")
